prog/learn_storage.rb

```diff
@@ -12,7 +12,8 @@ def self.parse(s)
   end
 
   label def start
-    q_storage_root = "/var".shellescape.freeze
+    q_storage_root = "/var/storage".shellescape.freeze
+    sshable.cmd("mkdir -p #{q_storage_root}")
```

Comment:
slightly weird to make "learning" things about a computer also have a side effect of creating a directory. Is it practical to have it somewhere else, e.g. prephost?

Reply:
Indeed, I had the same thought. However, we budd all strands simultaneously. Consequently, `LearnStorage` might execute before `Prog::Vm::PrepHost`.

```ruby
  label def prep
    bud Prog::Vm::PrepHost
    bud Prog::LearnNetwork unless vm_host.net6
    bud Prog::LearnMemory
    bud Prog::LearnCores
    bud Prog::LearnStorage
    bud Prog::InstallDnsmasq
    hop_wait_prep
  end
```

Reply:
Hmm. I think there are ways to resolve this, but none of them are quite short enough and our storage situation is too fluid for me to get too excited about adding even an extra ten lines of code to make it more explicit, and the downside risk of this change is mostly small.

Except, in one respect: I think if we do this, we need to start being more explicit about the ownership and mode bits on /var/storage.

Reply:
@pykello has another idea. We can check `/var/storage`, if it exists. So it will be there for mounted external disks. It makes sense to me.

```
if [ -d /var/storage ]; then df -B1 --output=size,avail /var/storage; else df -B1 --output=size,avail /var; fi
```

Reply:
Seems like a good idea.